uaclient/entitlements/tests/test_livepatch.py

```diff
@@ -43,59 +43,64 @@
+    def test_contract_status_entitled(self, livepatch_entitlement):
         """The contract_status returns ENTITLED when entitled is True."""
-        self.assertEqual(status.ENTITLED, self.entitlement.contract_status())
+        assert status.ENTITLED == livepatch_entitlement.contract_status()
 
-    def test_contract_status_unentitled(self):
+    def test_contract_status_unentitled(self, livepatch_entitlement):
         """The contract_status returns NONE when entitled is False."""
         livepatch_unentitled = copy.deepcopy(dict(LIVEPATCH_RESOURCE_ENTITLED))
 
         # Make livepatch resource access report not entitled
         livepatch_unentitled['entitlement']['entitled'] = False
-        self.cfg.write_cache('machine-access-livepatch', livepatch_unentitled)
-        self.assertEqual(status.NONE, self.entitlement.contract_status())
+        livepatch_entitlement.cfg.write_cache(
+            'machine-access-livepatch', livepatch_unentitled)
+        assert status.NONE == livepatch_entitlement.contract_status()
 
 
-class TestLivepatchOperationalStatus(BaseEnabledLivepatchConfig):
+class TestLivepatchOperationalStatus:
 
-    def test_operational_status_inapplicable_on_checked_affordances(self):
+    def test_operational_status_inapplicable_on_checked_affordances(
+            self, livepatch_entitlement):
         """The operational_status details failed check_affordances."""
         livepatch_bionic = copy.deepcopy(dict(LIVEPATCH_RESOURCE_ENTITLED))
         livepatch_bionic['entitlement']['affordances']['series'] = ['bionic']
-        self.cfg.write_cache('machine-access-livepatch', livepatch_bionic)
+        livepatch_entitlement.cfg.write_cache(
+            'machine-access-livepatch', livepatch_bionic)
 
         with mock.patch('uaclient.util.get_platform_info') as m_platform_info:
             m_platform_info.return_value = 'xenial'
-            op_status, details = self.entitlement.operational_status()
+            op_status, details = livepatch_entitlement.operational_status()
         assert op_status == status.INAPPLICABLE
         assert 'Livepatch is not available for Ubuntu xenial.' == details
 
-    def test_contract_status_unentitled(self):
+    def test_contract_status_unentitled(self, livepatch_entitlement):
         """The contract_status returns NONE when entitled is False."""
-        self.cfg.write_cache('machine-access-livepatch',
-                             {'entitlement': {'entitled': False}})
-        self.assertEqual(status.NONE, self.entitlement.contract_status())
+        livepatch_entitlement.cfg.write_cache(
+            'machine-access-livepatch', {'entitlement': {'entitled': False}})
+        assert status.NONE == livepatch_entitlement.contract_status()
 
 
-class TestLivepatchProcessDirectives(TestCase):
+class TestLivepatchProcessDirectives:
 
-    @params(('remoteServer', 'remote-server=%s'), ('caCerts', 'ca-certs=%s'))
+    @pytest.mark.parametrize('directive_key,livepatch_param_tmpl', (
```

Comment:
pytest allows us to generically "mark" tests with flags. These can be freeform for our own use, but there are a few flags that have a built-in meaning (e.g. @pytest.mark.skip will skip the decorated test), including `parametrize`, which specifies that this test should be run multiple times with the given parameters.

See https://docs.pytest.org/en/latest/mark.html for some pointers to the built-in marks, and https://docs.pytest.org/en/latest/example/markers.html#mark-examples for some information about (and examples of) custom marks.